read_tag_block.py

```diff
@@ -1,35 +1,85 @@
 from pycomm.ab_comm.clx import Driver as ClxDriver
 from nio.block.base import Block, Signal
-from nio.block.mixins.enrich.enrich_signals import EnrichSignals
+from nio.block.mixins import EnrichSignals, Retry
 from nio.properties import VersionProperty, StringProperty, Property
 
 
-class ReadTag(EnrichSignals, Block):
+class ReadTag(EnrichSignals, Retry, Block):
 
-    version = VersionProperty("0.1.4")
-    host = StringProperty(title='Host Address')
-    tags = Property(title='Tags')
+    host = StringProperty(title='Host Address', order=0)
+    tags = Property(title='Tags', order=1)
+    version = VersionProperty('0.2.0')
 
     def __init__(self):
         super().__init__()
-        self.cnxn = ClxDriver()
+        self.cnxn = None
+
+    def before_retry(self, *args, **kwargs):
+        self._disconnect()
+        self._connect()
 
     def configure(self, context):
         super().configure(context)
-        self.cnxn.open(self.host())
+        try:
+            self._connect()
+        except Exception:
+            self.cnxn = None
+            msg = 'Unable to connect to {}'.format(self.host())
+            self.logger.exception(msg)
 
     def stop(self):
         super().stop()
-        self.cnxn.close()
+        self._disconnect()
 
     def process_signals(self, signals):
+        host = self.host()
         output_signals = []
+        if self.cnxn is None:
+            try:
+                msg = 'Connecting to {}'.format(host)
+                self.logger.warning(msg)
+                self._connect()
+            except Exception as exc:
+                self.cnxn = None
+                msg = 'Unable to connect to {}'.format(host)
+                self.logger.error(msg)
+                raise exc
         for signal in signals:
-            value = self.cnxn.read_tag(self.tags(signal))
-            if not isinstance(value[0], tuple):
-                # read_tag only includes the tag's name in the return value
-                # when reading a list of tags, so we include it here
-                value = (self.tags(signal),) + value
-            new_signal = self.get_output_signal({'value': value}, signal)
-            output_signals.append(new_signal)
+            tag = self.tags(signal)
+            try:
+                value = self.execute_with_retry(self._make_request, tag)
+            except Exception as exc:
+                value = False
+                self.cnxn = None
+                msg = 'read_tag failed, host: {}, tag: {}'
+                self.logger.error(msg.format(host, tag))
+                raise exc
```

Comment:
Do we need to re-raise here? All that will do is dump a big stack trace from the block router. Since we've caught the error I would think we can just `self.logger.exception` and then `continue` or `break` out of the loop

Reply:
We may not even want to do any breaking/continuing since we're setting `value = False`. I would think if we're doing that we would expect to use `value` somewhere downstream from this

Reply:
Good point, this also applies to the EthernetIP blocks, so I'll make similar changes there.